<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="JWpublic">
<title>Orbicella Coral Fasta import • JWpublic</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Orbicella Coral Fasta import">
<meta property="og:description" content="JWpublic">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">JWpublic</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Avian-response_Landscape-recovery.html">Avian Response to Recovering Landscapes</a>
    <a class="dropdown-item" href="../articles/JW_Oribicella_fasta_import.html">Orbicella Coral Fasta import</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jwaterford/JW-Pages/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Orbicella Coral Fasta import</h1>
                        <h4 data-toc-skip class="author">James
Waterford</h4>
            
            <h4 data-toc-skip class="date">2023-06-08</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jwaterford/JW-Pages/blob/HEAD/vignettes/JW_Oribicella_fasta_import.Rmd" class="external-link"><code>vignettes/JW_Oribicella_fasta_import.Rmd</code></a></small>
      <div class="d-none name"><code>JW_Oribicella_fasta_import.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="transcriptomic-analysis-of-bleached-orbicella-faveolata">Transcriptomic analysis of bleached Orbicella Faveolata<a class="anchor" aria-label="anchor" href="#transcriptomic-analysis-of-bleached-orbicella-faveolata"></a>
</h3>
<p>Due to increasing ocean temperature, species such as coral have been
experiencing higher and higher levels of bleaching. The algal symbionts
that live with the coral are known to leave when temperatures persist at
even a few degrees higher than the normal water temperature. Bleaching
isn’t always permanent, however, and there is a brief amount of time in
which the algae can come back and restore life to the coral.</p>
<p>While we know this to be true, we still do not know specifically what
mediates the survival rates of these coral.</p>
<p>By analyzing the full transcriptome of both the coral and its
symbiont, I hope to better comprehend what genes are responsible for
this difference.</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Pinzón, Jorge H. et al. (2015), Data from: Whole transcriptome
analysis..during and after bleaching in a reef-building coral, Dryad,
Dataset, <a href="https://doi.org/10.5061/dryad.fj276" class="external-link uri">https://doi.org/10.5061/dryad.fj276</a></p>
<p>This data set has been retrieved from the DRYAD public domain, and
released by The University of Texas at Arlington, and Pennsylvania State
University.</p>
</div>
<div class="section level2">
<h2 id="reference-genome">Reference Genome<a class="anchor" aria-label="anchor" href="#reference-genome"></a>
</h2>
<p>The FASTA files have already been collected, so our next step is to
compare the reads (sequences along the transcriptome) to the full
genome.</p>
<p>To compare to the full genome, I need a reference. NCBI, and
Smithsonian Tropical Research Institute have created a full
representation of the Orbicella faveolata genome. “<a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_002042975.1/" class="external-link uri">https://www.ncbi.nlm.nih.gov/assembly/GCF_002042975.1/</a>”</p>
<p>As of 2017: 30,162 out of 32,574 sequences are mapped as genes or
psuedogenes.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">BiocManager</a></span><span class="op">)</span></span>
<span><span class="co"># install("Rsubread")</span></span></code></pre></div>
<div class="section level3">
<h3 id="mapping">Mapping<a class="anchor" aria-label="anchor" href="#mapping"></a>
</h3>
<p>Note: Due to difficulties with this large dataset, some data
processes have been removed/commented out. Specifically, the
<em>alignment</em> and <em>featurecount</em> processes.</p>
<p>I rely on Rsubread to map the reference genome to our samples. First,
I will build an index based on the reference genome. The purpose of
generating an index is to then compare the fasta reads to it.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/Rsubread" class="external-link">Rsubread</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">index</span> <span class="op">=</span> <span class="st">"GCF_002042975.1_ofav_dov_v1_genomic.fna"</span></span>
<span><span class="co">## buildindex(basename = "my_index", reference = index)</span></span>
<span></span>
<span><span class="va">treated_fasta</span> <span class="op">&lt;-</span> <span class="st">"ESM2_Bleaching_Metatranscriptome_transcriptome.fasta"</span></span>
<span><span class="va">wild_fasta</span> <span class="op">&lt;-</span> <span class="st">"ESM3_Bleaching_O-faveolata_transcriptome.fasta"</span></span>
<span></span>
<span></span>
<span><span class="co">## align(index = "my_index", readfile1=treated_fasta, output_file = "treatedResults.bam")</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="featurecounts">featureCounts<a class="anchor" aria-label="anchor" href="#featurecounts"></a>
</h3>
<p>To understand the genome better, we will add an annotation that
includes Gene Id and its general purpose.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ann</span> <span class="op">&lt;-</span> <span class="st">"genomic.gtf"</span></span>
<span></span>
<span><span class="co">## treatment_Count &lt;- featureCounts("treatedResults.bam", annot.ext = ann, isGTFAnnotationFile = TRUE)</span></span></code></pre></div>
<p>Mapping suggests something is wrong with my index. Not enough genes
are being mapped properly. But this is a great start!</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by James J. Waterford.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
